integration/e2e/metrics_generator_test.go

```diff
@@ -22,10 +22,110 @@ import (
 )
 
 const (
-	configMetricsGenerator           = "config-metrics-generator.yaml"
-	configMetricsGeneratorTargetInfo = "config-metrics-generator-targetinfo.yaml"
+	configMetricsGenerator                = "config-metrics-generator.yaml"
+	configMetricsGeneratorTargetInfo      = "config-metrics-generator-targetinfo.yaml"
+	configMetricsGeneratorMessagingSystem = "config-metrics-generator-messaging-system.yaml"
 )
 
+func TestMetricsGeneratorMessagingSystemLatencyHistogramEnabled(t *testing.T) {
+	s, err := e2e.NewScenario("tempo_e2e")
+	require.NoError(t, err)
+	defer s.Close()
+
+	// Use a config that enables the messaging system latency histogram
+	require.NoError(t, util.CopyFileToSharedDir(s, configMetricsGeneratorMessagingSystem, "config.yaml"))
+	tempoDistributor := util.NewTempoDistributor()
+	tempoIngester := util.NewTempoIngester(1)
+	tempoMetricsGenerator := util.NewTempoMetricsGenerator()
+	prometheus := util.NewPrometheus()
+	require.NoError(t, s.StartAndWaitReady(tempoDistributor, tempoIngester, tempoMetricsGenerator, prometheus))
+
+	isServiceActiveMatcher := func(service string) []*labels.Matcher {
+		return []*labels.Matcher{
+			labels.MustNewMatcher(labels.MatchEqual, "name", service),
+			labels.MustNewMatcher(labels.MatchEqual, "state", "ACTIVE"),
+		}
+	}
+	require.NoError(t, tempoDistributor.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"tempo_ring_members"}, e2e.WithLabelMatchers(isServiceActiveMatcher("ingester")...), e2e.WaitMissingMetrics))
+	require.NoError(t, tempoDistributor.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"tempo_ring_members"}, e2e.WithLabelMatchers(isServiceActiveMatcher("metrics-generator")...), e2e.WaitMissingMetrics))
+
+	c, err := util.NewJaegerGRPCClient(tempoDistributor.Endpoint(14250))
```

Comment:
The signature changed

```suggestion
	c, err := util.NewJaegerToOTLPExporter(tempoDistributor.Endpoint(14250))
```